packages/react-router/src/router.ts

```diff
@@ -768,6 +778,15 @@ export class Router<
 
     this.isServer = this.options.isServer ?? typeof document === 'undefined'
 
+    this.pathParamsDecodeCharMap = this.options.pathParamsAllowedCharacters
+      ? new Map(
+          this.options.pathParamsAllowedCharacters.map((char) => [
+            encodeURIComponent(char),
+            char,
+          ]),
+        )
+      : new Map()
```

Comment:
leave `pathParamsDecodeCharMap`  `undefined` in case `pathParamsAllowedCharacters` is not configured